src/merkle.rs

```diff
@@ -2,8 +2,15 @@ use sha3::{Digest, Sha3_256};
 
 type Hash = [u8; 32];
 
-struct Merkle {
-    tree: Vec<Hash>, // This will be a binary tree represented as a vector
+#[derive(Debug)]
+struct MerkleTree {
+    tree: Vec<Vec<Hash>>, // This will be a binary tree represented as a vector, with each level as a sub-vector with the hashes
+    leaves: Vec<Hash>,    // This will be uses to add new elements to the tree
```

Comment:
You don't need the leaves to add new elements, the hash of the elements will be the same, you can just use the last layer of the tree and add the new hash

Reply:
I think that having a leaves field in this struct avoids adding more lines and refactoring core functions. Do you think that is a MUST this feature?